cypress/support/commands.js

```diff
@@ -0,0 +1,149 @@
+import * as testData from '../fixtures/testData.json'
+import Selectors from "../support/selectors.js"
+
+Cypress.Commands.add('checkMainPageIsOpen', () => {
+    cy.url().should('eq', testData.carsMainPage)
+    cy.get(Selectors.logo).should('be.visible')
+})
+
+Cypress.Commands.add('goToResearchReviewsPage', () => {
+    cy.get(Selectors.pageMenuTitles).contains(testData.testResearchLinkTitle).click()
+})
+
+Cypress.Commands.add('checkResearchReviewsPageIsOpen', () => {
+    cy.url().should('contain', testData.researchPage)
+    cy.get(Selectors.researchHeading).contains(testData.testResearchLinkTitle)
+})
+
+Cypress.Commands.add('selectRandomOption', (optionSelector, menuSelector) => {
+    function getRandomInt(min, max) {
+        return Math.floor(Math.random() * (max - min + 1)) + min
+    }
+    return cy.get(optionSelector).then(listing => {
+        const randomNumber = getRandomInt(0, listing.length - 1)
+        return cy.get(optionSelector).eq(randomNumber).invoke('text').then(text => {
+            return cy.get(menuSelector).select(text).then(() => {
+                return text
+            })
+        })
+    })
+})
+
+Cypress.Commands.add('selectRandomCharacteristics', (fileName) => {
+    const options = {}
+
+    cy.selectRandomOption(Selectors.makeMenuOption, Selectors.makeMenu)
+        .then(text => {
+            options.make = text
+            return text
+        })
+        .then(() => cy.selectRandomOption(Selectors.modelMenuOption, Selectors.modelMenu))
+        .then(text => {
+            options.model = text
+            return text
+        })
+        .then(() => cy.selectRandomOption(Selectors.yearMenuOption, Selectors.yearMenu))
+        .then(text => {
+            options.year = text
+            return text
+        })
+        .then(() => {
+            const filePath = 'cypress/fixtures/' + fileName
+            cy.writeFile(filePath, options)
+        })
+})
+
+Cypress.Commands.add('checkOptionsAreSelected', () => {
+    cy.get(Selectors.noOptionSelectedInMakeMenu).should('not.exist')
+})
+
+Cypress.Commands.add('clickResearchButton', () => {
+    cy.get(Selectors.researchButton).click()
+})
+
+Cypress.Commands.add('checkPageWithCarDescriptionOpens', (filePath) => {
+    cy.fixture(filePath).then(options => {
+        const selector = `${Selectors.selectedOptionsTitle1}:contains(${options.make}), ${Selectors.selectedOptionsTitle2}:contains(${options.make})`
+        cy.get(selector).should('exist').and('be.visible')
+    })
+})
+
+Cypress.Commands.add('selectBasicTrim', () => {
+    cy.get(Selectors.trimOptions).children().eq(0).find('a').click()
+})
+
+Cypress.Commands.add('storeCarCharacteristics', (fileName) => {
```

Comment:
Saving data to a file looks complex. 
You can `return cy.wrap(carInfo)`
In tests you would use it like 
```
cy.storeCarCharacteristics.then((carInfo) => {
})
```

Same goes to command selectRandomCharacteristics